### ISSUE TITLE
Make menu font size configurable

### ISSUE THREAD

**Issue by glistok** [arduino/arduino-ide#1099]
### Describe the request

Please add a setting to configure the font size in the menus.

### Describe the current behavior

The current size is too small for me to read. It needs to be in the range of font size 20 to 24.

### Arduino IDE version

2

### Operating system

Windows

### Operating system version

11

#### Additional requests

- https://github.com/arduino/arduino-ide/issues/1099#issuecomment-2142863940
- https://forum.arduino.cc/t/font-settings-for-the-ide-menus/1378827
- https://forum.arduino.cc/t/font-settings-for-the-ide-menus/1378827/1
- https://forum.arduino.cc/t/font-settings-for-the-ide-menus/1378827/11

#### Related

- https://github.com/arduino/arduino-ide/issues/1495

<a name="workaround"></a>

#### Workaround

You can control the scale of the entire Arduino IDE user interface using the `--force-device-scale-factor` flag.

This is different from Arduino IDE's **Edit > Increase Font Size** / **Decrease Font Size** feature (when the "**Interface scale: ☐ Automatic**" preference is enabled) because that feature scales up all UI components other than the menu bar and menus, while `--force-device-scale-factor` scales up the entire UI. This means you can use both features together to increase the menu font size by first using `--force-device-scale-factor` to scale up the size of the entire UI, then **Decrease Font Size** to scale down the size of the rest of the UI:

1. Select **File > Quit** from the Arduino IDE menus if it is running.
   All Arduino IDE windows will close.
1. Open a terminal in the Arduino IDE installation folder:
   - [**Linux**](https://ubuntu.com/tutorials/command-line-for-beginners).
   - [**macOS**](https://support.apple.com/guide/terminal/apd5265185d-f365-44cb-8b09-71a064a42125/mac)
   - [**Windows**](https://learn.microsoft.com/powershell/scripting/windows-powershell/starting-windows-powershell?view=powershell-7.3#from-the-start-menu).
1. Type the following command a the command prompt:
   - **Windows**:
     ```text
     & ".\Arduino IDE" --force-device-scale-factor=<factor>
   - **Linux**:
     ```text
     ./arduino-ide --force-device-scale-factor=<factor>
     ```
   - **macOS**:
     ```text
     "/Applications/Arduino IDE.app/Contents/MacOS/Arduino IDE" --force-device-scale-factor=<factor>
     ```
1. Replace the `<factor>` placeholder in the command with the factor by which you want to increase the scale of the menu text.
   **ⓘ** For example, if you wanted to increase the scale of the menu by a factor of 1.5, you would replace the `<factor>` placeholder with `1.5`.
1. Press the <kbd>**Enter**</kbd> key.
1. Wait for Arduino IDE to finish starting.
1. Select **File > Preferences...** (or **Arduino IDE > Settings...** for macOS users) from the Arduino IDE menus.
   The "**Preferences**" dialog will open.
1. Check the box next to "**Interface scale: ☐ Automatic**" in the "**Preferences**" dialog.
   **ⓘ** This setting causes the **Edit > Increase Font Size** / **Decrease Font Size** feature to scale the UI instead of only changing the font size.
1. Click the "**OK**" button.
   The "**Preferences**" dialog will close.
1. Select **Edit > Decrease Font Size** from the Arduino IDE menus.
   The scale of the Arduino IDE user interface will decrease, but the scale of the menu bar will remain the same.
1. Repeat the previous step until the scale of the rest of the Arduino IDE user interface is according to your preferences.

Once you have found your preferred scale factor, for the sake of convenience you can create a shortcut that starts Arduino IDE with the appropriate `--force-device-scale-factor` flag.

### Issue checklist

- [X] I searched for previous requests in [the issue tracker](https://github.com/arduino/arduino-ide/issues?q=)
- [X] I verified the feature was still missing when using the latest [nightly build](https://github.com/arduino/arduino-ide#nightly-builds)
- [X] My request contains all necessary details

**Comment by VOC-LINK** [comment]
Listen, the tiny menu letters of Arduino IDE 1.8.19 are **completely unreadable**! They should have twice of their current size for today — for presently used screen resolutions — and actually there should be an option allowing resizing them at will! What does it mean „it cannot be done”? Hire better programmers if the current ones are unable to fix that, simple.
I don't want to sound rude, but I found the answer „it cannot be done” as both ridiculous and rude; it simply means „get lost, we don't care, buy better glasses or something”. Never been fond of Java, but even when creating programs in that atrocity, don't you tell me it's not possible to set the font size.

**Comment by per1234** [comment]
Hi @VOC-LINK. I guess you missed the news that Arduino IDE 1.x was replaced by a complete rewrite of the application 1.5 years ago:

https://blog.arduino.cc/2022/09/14/its-here-please-welcome-arduino-ide-2-0/

This is the issue tracker for that Arduino IDE 2.x codebase so reports of problems with Arduino IDE 1.8.19 are off topic here. Arduino IDE 1.8.19 is outdated and the Arduino IDE 1.x codebase is no longer maintained so we are not accepting any bug reports for that codebase. Please update to the modern Arduino IDE. You can get it from the download links listed here:

https://www.arduino.cc/en/software

If for some reason you can't or won't update, you can request assistance from the community with using the outdated version of Arduino IDE on Arduino Forum:

https://forum.arduino.cc/c/using-arduino/ide-1-x/18
